engine/new_runtime/src/noncontiguous_memory.rs

```diff
@@ -0,0 +1,20 @@
+use crate::file_memory::FileMemory;
+use crate::ram_memory::RamMemory;
+use crate::locked_memory::MemoryConfiguration;
+use crate::types::Bytes;
+use zeroize::Zeroize;
+
+// NONCONTIGUOUS MEMORY
+/// Shards of memory which composes a non contiguous memory
+enum MemoryShard<T: Zeroize + Bytes> {
+    File(FileMemory),
+    Ram(RamMemory<T>)
+}
+
+// We set the maximum number of shards to 8, this is an arbitrary value that can be discussed
+const MAX_SHARDS: usize = 8;
+
+pub struct NonContiguousMemory<T: Zeroize + Bytes> {
```

Comment:
`MAX_SHARDS` could also be implemented via const generics.

Reply:
I've removed the dynamic number of shards and just use 2 shards because it does not really improve security to have more than 2 shards